kotlinx-coroutines-core/jvm/test/channels/RendezvousChannelStressLFTest.kt

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.coroutines.channels
+
+import kotlinx.atomicfu.*
+import kotlinx.coroutines.*
+import java.util.concurrent.atomic.AtomicLong
+import java.util.concurrent.atomic.AtomicLongArray
+import kotlin.math.*
+import kotlin.test.*
+
+/**
+ * Tests lock-freedom of send and receive operations on rendezvous channel.
+ * There is a single channel with two sender and two receiver threads.
+ * When one sender or receiver gets suspended at most one other operation is allowed to cease having progress
+ * (`allowSuspendedThreads = 1`).
```

Comment:
It would be nice to have the ability to specify "at most X suspended thread **in this particular thread group**" to suspend both senders and receivers.
Not the review comment, but as a future improvement